test/e2e/test/helper/yaml.go

```diff
@@ -353,8 +353,29 @@ func transformToE2E(namespace, fullTestName, suffix string, transformers []Build
 			decodedObj.Namespace = namespace
 			decodedObj.Name = decodedObj.Name + "-" + suffix
 		case *appsv1.DaemonSet:
+			name := decodedObj.Name + "-" + suffix
 			decodedObj.Namespace = namespace
-			decodedObj.Name = decodedObj.Name + "-" + suffix
+			decodedObj.Name = name
+			decodedObj.Spec.Selector.MatchLabels["name"] = name
+			decodedObj.Spec.Template.ObjectMeta.Labels["name"] = name
+			for i, init := range decodedObj.Spec.Template.Spec.InitContainers {
```

Comment:
deserves a comment that this is only for the non-root recipe, also probably worth pulling in separate function